### PR TITLE
New Literature Requests - Details page and fixes

### LINKED ISSUE #334: New literature request CRUD
Librarian should be able to CRUD (Create/Update/Delete) new literature request.

- [x] create a new edit page with the UI web form in the backoffice for new literature request
- [x] make sure backend permission are correct: librarian can read/list/edit/create/delete, patron can only read/list its own requests (not all) 
- [x] **optional** maybe a good idea to add the Steps in the UI on the top as for ILL/ACQ

Make sure that the librarian can select/change which patron created the request and attach/replace document, ILL/ACQ.

### LINKED ISSUE #356: frontsite literature request form: changes
change description of fields:
- [ ] "Payment method" -> "Payment method (only for Purchase)"
- [ ] "Payment information" -> "Payment information (Only for purchase)"

### PR DESCRIPTION
**Pull request by ntarocco**
* closes #334
* closes #356

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -101,7 +101,7 @@
     "react-router-dom": "^5.1.2",
     "react-scripts": "^3.4.1",
     "react-scroll": "^1.7.16",
-    "react-searchkit": "^1.0.0-alpha.11",
+    "react-searchkit": "^1.0.0-alpha.14",
```
Comment (ntarocco) [inline package.json:104]: Remember to add this also in CDS-ILS

**Comment by ntarocco** [comment]
Screenshot 2021-02-19 at 20 05 29
Screenshot 2021-02-19 at 20 06 37
Screenshot 2021-02-19 at 20 06 31
Screenshot 2021-02-19 at 20 06 22
Screenshot 2021-02-19 at 20 06 12
Screenshot 2021-02-19 at 20 05 51

File: src/lib/pages/frontsite/DocumentRequests/DocumentRequestForm.js [hunk 2]
```diff
@@ -299,12 +299,12 @@ DocumentRequestForm.defaultProps = {
     placeholder: 'Select type...',
   },
   paymentMethod: {
-    label: 'Payment method',
+    label: 'Payment method (if purchase)',
```
Comment (kprzerwa) [inline src/lib/pages/frontsite/DocumentRequests/DocumentRequestForm.js:302]: it came out actually in the user testing too :)

**Review by kprzerwa — APPROVED** [review]
thanks for the screenshots, discord distorts the colors for me
LGTM the code
the UX improvements 🚀

**Pull request merged**